### ISSUE TITLE
Idea: Allow turning on using conventions to build queue names

### ISSUE THREAD

**Issue by gabrieljoelc** [jondot/sneakers#164]
It'd be nice if there was a way to not _have_ to call `from_queue` for every worker and use convention instead. We have done something like this in [la_gear](https://github.com/giftcardzen/la_gear), but could this be something for sneakers? Here's what we do in la_gear [lines](https://github.com/giftcardzen/la_gear/blob/master/lib/la_gear/worker.rb#L27-L53) and these [lines](https://github.com/giftcardzen/la_gear/blob/master/lib/la_gear/worker.rb#L7-L9):
```ruby
# lib/la_gear/worker.rb
def self.included(base)
  # some other stuff...
  base.send(:include, ::Sneakers::Worker)
  base.extend(SneakersClassMethods)
  base.from_queue base.default_queue_name, base.default_queue_opts
  # some other stuff...
end

# some other stuff...

module SneakersClassMethods
  def default_queue_name
    "#{::Sneakers::CONFIG.fetch(:app_name, 'sneakers_app').underscore}.#{routing_key}"
  end

  def default_queue_opts
    {
      routing_key: [routing_key, retry_routing_key],
      arguments: default_queue_args,
      handler_opts: {
        routing_key: retry_routing_key
      }
    }
  end

  def routing_key
    name.underscore.gsub('/', '.')
  end

  def retry_routing_key
    "#{default_queue_name}.retry"
  end

  def default_queue_args
    { 'x-dead-letter-exchange' => "#{::Sneakers::CONFIG.fetch(:exchange, 'sneakers').underscore}.retry" }
  end
end
```
In some of our workers, we override the convention by just calling `from_queue` again after the `include LaGear::Worker` with our overrides (i.e. 2 workers in the same sneakers process/app that subscribe to the same message).

The routing key convention in the `routing_key` class method above allows us to do some fun things like namespacing "versioned" worker classes for supporting backwards compatibility when we make changes to existing workers but still want our backlog of production messages to be processed after the "version" of the worker goes live. For example, this worker (see [here](https://github.com/giftcardzen/la_gear/blob/master/test/test_worker.rb#L6-L10) and [here](https://github.com/giftcardzen/la_gear/blob/master/test/test_worker.rb#L77-L79)):
```ruby
module NotPogs
  class V2
    include LaGear::Worker
  end
end
```
will create a queue with `not_pogs.v2` as the routing key.

As an aside, the `Sneakers::CONFIG[:app_name]` thing is for allowing us to use a single direct exchange so we can have _multiple_ sneakers process with workers in process A subscribing to the same routing keys as workers in process B.

**Comment by jondot** [comment]
Yup, I've seen many developers use this kind of thing. I realize now that the name of a queue might not have to be explicit - though internally we usually see the verboseness of specifying a queue each time as good self-documenting code.
I'd welcome any pull request that implements this idea as long as it also welcomes the explicit way of saying `from_queue`. 

Would that be fine from your perspective?
